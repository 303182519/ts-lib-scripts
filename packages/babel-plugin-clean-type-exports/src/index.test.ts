import { transform } from '@babel/core';
import { join } from 'path';
import plugin from './index';

it('验证变量名变更', () => {
  const input = `
    import { B, C, D, E } from './E';
    x === y;
    console.log(B);
    export {
      /* type */
      B,
      /* type */
      C, 
      // type
      D,
      E
    };
  `;

  const result = transform(input, {
    filename: 'test.ts',
    plugins: [plugin],
    babelrc: false,
    cwd: join(__dirname, '.'),
    presets: [
      [
        '@babel/preset-env',
        {
          modules: false,
        },
      ],
    ],
  });

  expect(result?.code).toMatchSnapshot();
});
