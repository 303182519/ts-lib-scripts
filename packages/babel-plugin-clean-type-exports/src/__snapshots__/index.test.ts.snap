// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`验证变量名变更 1`] = `
"import { B, C, D, E } from './E';
x === y;
console.log(B);
export { E };"
`;
